### PR TITLE
Proxy cleanup

### PR DESCRIPTION
**Pull request by thockin**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Refactor a bit of proxy setup code to make future changes easier.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -576,6 +576,57 @@ func createClients(config componentbaseconfig.ClientConnectionConfiguration, mas
...
+	}
+
+	fn := func() {
+		err := hz.Run()
+		if err != nil {
+			// For historical reasons we do not abort on errors here.  We may
+			// change that in the future.
+			klog.Errorf("healthz server failed: %v", err)
+		} else {
+			klog.Errorf("healthz server returned without error")
```
Comment (aojea) [inline cmd/kube-proxy/app/server.go]: is this an Error or should be `klog.V(3).Infof(`?

Comment (thockin) [reply cmd/kube-proxy/app/server.go]: I think it's an error - it's not supposed to return.

**Comment by aojea** [comment]
/cc @danwinship 
he was dealing with the bugs in the health server recently

**Review by SataQiu — COMMENTED** [review]
/lgtm
/hold
Hold for more reviews. Feel free to cancel hold if needed.

**Comment by thockin** [comment]
This SHOULD be just a cleanup, not much behavior change.  The only change is that healthz will retry more frequently.

**Pull request merged**